#AUTOGENERATED! DO NOT EDIT! File to edit: dev/01_data.ipynb (unless otherwise specified).

__all__ = ['download_file', 'get_files', 'num_files', 'url', 'get_pretrained_models', 'xceptionURL']

#Cell
from ..imports import *
import requests
import urllib

#Cell
def download_file(url, dest):
    if os.path.exists(dest): raise Exception(f"path {dest} already exists.")
    s = requests.Session()
    u = s.get(url, stream=True)
    with open(dest, "wb") as f:
        nbytes = 0
        for chunk in u.iter_content(chunk_size=1024*1024):
            nbytes += len(chunk)
            f.write(chunk)

#Cell
def get_files(path):
    "return a list of file names present in `path`"
    return [o.name for o in os.scandir(path)]

#Cell
def num_files(path):
    "return number of files in `path`"
    return len(os.listdir(path))

#Cell
url  = 'http://kaldir.vc.in.tum.de/FaceForensics/models/faceforensics++_models.zip'

#Cell
def get_pretrained_models(url, dest):
    "download files in url at dest"
    return urllib.request.urlretrieve(url, f"./{dest}")

#Cell
xceptionURL = "https://data.lip6.fr/cadene/pretrainedmodels/xception-b5690688.pth"